pages/api/proxy/transcript/index.ts

```diff
@@ -0,0 +1,137 @@
+import convertVttToJson from '@lib/convert/string/convertVttToJson';
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<string | { error: IRssProxyError }>
+) {
+  const { u, cb } = req.query;
+  const transcriptUrl = Array.isArray(u) ? u[0] : u;
+  const cacheBuster = Array.isArray(cb) ? cb[0] : cb;
+
+  try {
+    const fetchUrl = new URL(transcriptUrl);
+
+    if (cacheBuster) {
+      fetchUrl.searchParams.set('cb', cacheBuster);
+    }
+
+    const transcriptResponse = await fetch(fetchUrl.toString());
+
+    if (!transcriptResponse?.ok) {
+      throw new Error(
+        `URL provided return status "${transcriptResponse.statusText}"`
+      );
+    }
+
+    const transcriptAsText = await transcriptResponse.text();
+    const contentType =
+      transcriptResponse.headers.get('Content-Type') || 'text/plain';
+    const conversions: TranscriptTypeConversion<IRssPodcastTranscriptJson>[] = [
+      {
+        check: (ct: string, t: string) =>
+          /(?:application|text)\/json/i.test(ct) || t.startsWith('{'),
+        convert: (t: string) => JSON.parse(t)
+      },
+      {
+        check: (ct: string, t: string) =>
+          /(?:application|text)\/vtt/i.test(ct) || t.startsWith('WEBVTT'),
+        convert: (t: string) => convertVttToJson(t)
+      },
+      {
+        check: (ct: string, t: string) =>
+          /(?:application|text)\/srt/i.test(ct) || t.includes('-->'),
+        convert: (t: string) => convertVttToJson(convertSrtToVtt(t))
+      }
+    ];
+    const jsonResponseBody = conversions
+      .find(({ check }) => check(contentType, transcriptAsText))
+      ?.convert(transcriptAsText);
+    const hasMultipleSpeakers =
+      jsonResponseBody?.segments.filter(({ speaker }) => !!speaker).length > 1;
+    const speakerSegmentsBlocks = jsonResponseBody?.segments
+      .reduce((a, segment) => {
+        const aClone = [...a];
+        const previousSegment = aClone.pop();
+        const isSpace = segment.body.trim().length === 0;
+
+        if (
+          !previousSegment ||
+          isSpace ||
+          segment.startTime > previousSegment.endTime
+        ) {
+          return [...a, segment];
+        }
+
+        const updatedSegment = {
+          ...previousSegment,
+          body: previousSegment.body + segment.body
+        };
+
+        return [...aClone, updatedSegment];
+      }, [] as IRssPodcastTranscriptJsonSegment[])
+      .reduce((a, segment) => {
+        const aClone = [...a];
+        const previousBlock = aClone.pop();
+        const lastSegment = previousBlock?.segments.at(
+          previousBlock.segments.length - 1
+        );
+        const speakerChanged = previousBlock?.speaker !== segment.speaker;
+        const sentenceEnded = /[.?!]$/.test(lastSegment?.body || '');
+        const breakSegment =
+          (hasMultipleSpeakers && speakerChanged) ||
+          (!hasMultipleSpeakers && sentenceEnded);
```

Comment:
I just figured it out. This endpoint is tailored to what worked best to render transcripts in our app designs, and give us the flexibility to adapt the logic to accommodate changes later. Also was kind of necessary since there isn't a standard around combining cues that spans all the transcript data types this endpoint supports as input. Had to play it by ear to get the result that felt best.